sonar-plugin-api/src/main/java/org/sonar/api/batch/scm/ScmProvider.java

```diff
@@ -64,4 +64,19 @@ public BlameCommand blameCommand() {
   public Set<Path> branchChangedFiles(String targetBranchName, Path rootBaseDir) {
     return null;
   }
+
+  /**
+  * The relative path to the project from SCM root, typically ".".
+  */
+  public String projectBasedir() {
+    return ".";
+  }
+
+  /**
+   * The corresponding revision ID of the pull request,
```

Comment:
Why "of the pull request"? I would say: "The current revision ID of the analyzed code"